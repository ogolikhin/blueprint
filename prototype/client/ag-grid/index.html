<html>
<head>
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
  <meta http-equiv="content-type" content="text/html;charset=utf-8" />
  <meta content="en" http-equiv="content-language" />
  <meta name="viewport" content="initial-scale=1, maximum-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />

  <title>ag-grid</title>
  <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css" rel="stylesheet">
  <link href='https://fonts.googleapis.com/css?family=Open+Sans' rel='stylesheet' type='text/css'>

  <!-- <script src="https://code.jquery.com/jquery-1.12.2.js"></script> -->
  <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.3.8/angular.min.js"></script>
  <script src="dragdrop.js"></script>
  <script src="ag-grid.js"></script>
  <script src="app.js"></script>

  <style>
    html, body {
      margin: 0;
      padding: 0;
      height: 100%;
      font-family: "Open Sans", sans-serif;
    }
    #top-bar {
      line-height: 50px;
      height: 50px;
    }
    #main-container {
      height: calc(100% - 50px);
    }
    #centerPane {
      width: calc(50% - 10px);
      background: #cecece;
      color: #fff;
      position: relative;
    }
    #leftPane, #rightPane {
      width: calc(25% - 5px);
    }
    #rightPane {
      background: #fff;
    }

    /* styles related to the ree */
    .ag-body {
      background-color: #474747;
      color: #fff;
      padding: 0 !important;
    }
    .myrow {
      font: normal 14px "Open Sans", sans-serif;
      line-height: 30px;
    }
    .myicon {
      width: 30px;
      display: inline-block;
      text-align: center;
    }
    .mychild2 {
      padding-left: 26px;
    }
    .mychild {
      padding-left: 55px;
    }
    .ag-row-level-2 .mychild {
      padding-left: 81px;
    }
    .myhightlight {
      font-weight: bold;
      background: yellow;
      color: black;
    }
    .mychild.word::before {
      content: "\f1c2";
      font: normal normal normal 14px/1 FontAwesome;
      font-size: inherit;
      margin-left: -30px;
      width: 30px;
      display: inline-block;
      text-align: center;
      color: lightblue;
    }
    .mychild.excel::before {
      content: "\f1c3";
      font: normal normal normal 14px/1 FontAwesome;
      font-size: inherit;
      margin-left: -30px;
      width: 30px;
      display: inline-block;
      text-align: center;
      color: green;
    }
    .myrow.ag-row-focus {
      border-radius: 4px;
      background: #3e3e3e;
    }
    .mychild.ag-cell-focus {
      border-radius: 4px;
      background: #019cde !important;
      color: #fff !important;
    }
    .mychild.ag-cell-focus.word::before,
    .mychild.ag-cell-focus.excel::before {
      color: #fff;
    }

    .over {
      background-color: darkred;
    }

    /* styles related to accordion */
    #rightPane .utility-panel {
      background: #fff;
      padding: 0;
      vertical-align: bottom;
      height: 100%;
      overflow: hidden;
    }

    #rightPane .no-select, #rightPane h2 {
      -webkit-tap-highlight-color: transparent;
      -webkit-touch-callout: none;
      -webkit-user-select: none;
      -khtml-user-select: none;
      -moz-user-select: none;
      -ms-user-select: none;
      user-select: none;
    }

    #rightPane ul {
      list-style: none;
      padding: 0;
      margin: 0;
      height: 100%;
    }
    #rightPane ul li {
      position: relative;
      padding: 0;
      margin: 0;
      width: 100%;
    }
    #rightPane ul li input[type=checkbox].pin {
      z-index: 2;
      position: absolute;
      top: 5px;
      right: 24px;
      visibility: hidden;
    }
    #rightPane ul li input[type=radio].state {
      position: absolute;
      cursor: pointer;
      width: 100%;
      height: 33px;
      z-index: 1;
      opacity: 0;
    }
    #rightPane ul li i {
      position: absolute;
      transform: translate(-6px, 0);
      margin-top: 10px;
      right: 8px;
    }
    #rightPane ul li i:before, #rightPane ul li i:after {
      content: "";
      position: absolute;
      background-color: #fff;
      width: 3px;
      height: 9px;
    }
    #rightPane ul li h2.header {
      font-size: 16px;
      line-height: 32px;
      font-weight: 300;
      display: block;
      background-color: silver;
      margin: 0;
      cursor: pointer;
      color: #fff;
      padding-left: 4px;
      border-top: solid 1px black;
    }
    #rightPane ul li div.content {
      font-size: 14px;
      overflow: auto;
      margin: 0;
      height: 0;
      opacity: 0;
      z-index: 2;
    }

    #rightPane ul li input[type=radio].state:checked ~ input[type=checkbox].pin,
    #rightPane ul li input[type=checkbox].pin:checked {
      visibility: visible;
    }

    #rightPane ul li input[type=radio].state:checked ~ div,
    #rightPane ul li input[type=checkbox].pin:checked ~ div {
      opacity: 1;
      padding: 0 10px;
    }
    #rightPane ul li i:before {
      transform: translate(2px, 0) rotate(45deg);
    }
    #rightPane ul li i:after {
      transform: translate(-2px, 0) rotate(-45deg);
    }
    #rightPane ul li input[type=checkbox].pin:checked ~ i:before,
    #rightPane ul li input[type=radio].state:checked ~ i:before {
      transform: translate(-2px, 0) rotate(45deg);
    }
    #rightPane ul li input[type=checkbox].pin:checked ~ i:after,
    #rightPane ul li input[type=radio].state:checked ~ i:after {
      transform: translate(2px, 0) rotate(-45deg);
    }
    /* SCSS function to generate rule for equally dividing the height amongst siblings
	ref: http://stackoverflow.com/questions/8720931/can-css-detect-the-number-of-children-an-element-has */
    /*
		@for $i from 1 through 20 {
			li:first-child:nth-last-child( #{$i} ),
			li:first-child:nth-last-child( #{$i} ) ~ li {
				height: calc(100% / #{$i} - 10px);
			}
		}
	*/

    /* styles related to panes splitter */
    .split {
      -webkit-box-sizing: border-box;
      -moz-box-sizing: border-box;
      box-sizing: border-box;

      overflow-y: auto;
      overflow-x: hidden;
    }

    .gutter {
      background-color: #eee;

      background-repeat: no-repeat;
      background-position: 50%;
    }

    .gutter.gutter-horizontal {
      background-image: url('grips/vertical.png');
      cursor: ew-resize;
    }

    .gutter.gutter-vertical {
      background-image: url('grips/horizontal.png');
      cursor: ns-resize;
    }

    .split.split-horizontal, .gutter.gutter-horizontal {
      height: 100%;
      float: left;
    }
  </style>
</head>
<body ng-app="example" ng-controller="exampleCtrl">
<div id="top-bar" style="overflow: hidden">
  Blueprint top bar
  - <a href="javascript:void(0)" onclick="closeLeft()">[close left]</a> | <a href="javascript:void(0)" onclick="openLeft()">[open left]</a>
  - <a href="javascript:void(0)" onclick="closeRight()">[close right]</a> | <a href="javascript:void(0)" onclick="openRight()">[open right]</a>
</div>
<div id="main-container">
  <div id="leftPane" ag-grid="gridOptions" class="mygrid split split-horizontal"></div>
  <div id="centerPane" ui-on-drop="onDrop($data)" droppable class="split split-horizontal">Drop zone</div>
  <div id="rightPane" class="split split-horizontal"><div class="utility-panel">
    <ul bp-accordion header-height="33px">
      <li>
        <input type="radio" name="accordion" checked="checked" class="state"><input type="checkbox" class="pin">
        <i></i>
        <h2 class="header">Discussion</h2>
        <div class="content">This page was written in HTML and CSS. The CSS was compiled from SASS. I used Normalize as my CSS reset and -prefix-free to save myself some headaches. I haven't quite gotten the hang of Slim for compiling into HTML, but someday I'll use it since its syntax compliments that of SASS. Regardless, this could all be done in plain HTML and CSS.</div>
      </li>
      <li>
        <input type="radio" name="accordion" class="state"><input type="checkbox" class="pin">
        <i></i>
        <h2 class="header">Attachment</h2>
        <div class="content">Using the sibling and checked selectors, we can determine the styling of sibling elements based on the checked state of the checkbox input element. One use, as demonstrated here, is an entirely CSS and HTML accordion element. Media queries are used to make the element responsive to different screen sizes. Using the sibling and checked selectors, we can determine the styling of sibling elements based on the checked state of the checkbox input element. One use, as demonstrated here, is an entirely CSS and HTML accordion element. Media queries are used to make the element responsive to different screen sizes. Using the sibling and checked selectors, we can determine the styling of sibling elements based on the checked state of the checkbox input element. One use, as demonstrated here, is an entirely CSS and HTML accordion element. Media queries are used to make the element responsive to different screen sizes.</div>
      </li>
      <li>
        <input type="radio" name="accordion" class="state"><input type="checkbox" class="pin">
        <i></i>
        <h2 class="header">Relationships</h2>
        <div class="content">By making the open state default for when :checked isn't detected, we can make this system accessable for browsers that don't recognize :checked. The fallback is simply an open accordion. The accordion can be manipulated with Javascript (if needed) by changing the "checked" property of the input element.</div>
      </li>
      <li>
        <input type="radio" name="accordion" class="state"><input type="checkbox" class="pin">
        <i></i>
        <h2 class="header">Another item</h2>
        <div class="content">
          <input type="text" size="20" style="width: 250px"><br><br>
          <textarea rows="10" style="width: 250px"></textarea><br><br>
          <select multiple size="5" style="width: 250px">
            <option>Option 1
            <option>Option 2
            <option>Option 3
            <option>Option 4
            <option>Option 5
            <option>Option 6
          </select><br><br>
          <select style="width: 250px">
            <option>Option 1
            <option>Option 2
            <option>Option 3
            <option>Option 4
            <option>Option 5
            <option>Option 6
          </select>
        </div>
      </li>
    </ul></div></div>
</div>
</body>
<script src="split.js"></script>
<script>
  Split(['#leftPane', '#centerPane', "#rightPane"], {
    sizes: [25, 50, 25],
    minSize: 200
  });

  function parseWidth(str) {
    var regex = /\d+(\.\d+)?%/g;

    var floats = str.match(regex).map(function(v) { return parseFloat(v); });

    return floats.join("");
  }

  var leftOpen = true;
  var rightOpen = true;

  function closeLeft() {
    document.getElementById('leftPane').style.width = "0";
    if(rightOpen) {
      var width = document.getElementById('rightPane').style.width;
      width = 100 - parseFloat(parseWidth(width));
      width = "calc(" + width + "% - 15px)";
      document.getElementById('centerPane').style.width = width;
    } else {
      document.getElementById('centerPane').style.width = "calc(100% - 20px)";
    }
    leftOpen = false;
    return false;
  }

  function openLeft() {
    document.getElementById('leftPane').style.width = "calc(25% - 5px)";
    if(rightOpen) {
      var width = document.getElementById('rightPane').style.width;
      width = 75 - parseFloat(parseWidth(width));
      width = "calc(" + width + "% - 10px)";
      document.getElementById('centerPane').style.width = width;
    } else {
      document.getElementById('centerPane').style.width = "calc(75% - 15px)";
    }
    leftOpen = true;
    return false;
  }

  function closeRight() {
    document.getElementById('rightPane').style.width = "0";
    if(leftOpen) {
      var width = document.getElementById('leftPane').style.width;
      width = 100 - parseFloat(parseWidth(width));
      width = "calc(" + width + "% - 15px)";
      document.getElementById('centerPane').style.width = width;
    } else {
      document.getElementById('centerPane').style.width = "calc(100% - 20px)";
    }
    rightOpen = false;
    return false;
  }

  function openRight() {
    document.getElementById('rightPane').style.width = "calc(25% - 5px)";
    if(leftOpen) {
      var width = document.getElementById('leftPane').style.width;
      width = 75 - parseFloat(parseWidth(width));
      width = "calc(" + width + "% - 10px)";
      document.getElementById('centerPane').style.width = width;
    } else {
      document.getElementById('centerPane').style.width = "calc(75% - 15px)";
    }
    rightOpen = true;
    return false;
  }
</script>
</html>
