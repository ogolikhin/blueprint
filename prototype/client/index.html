<!DOCTYPE html>
<html>
  <head>
    <base href="/">
    <title>pegah's prototypes</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width">
    <link rel="stylesheet" href="bower_components/bootstrap/dist/css/bootstrap.css">
  </head>
  <body>
    <h3>Base Expreience/Nova Mockups</h3>
    <h4>
        <ul>
            <li><a href="kendo.html">Using Kendo UI (spliter/tree/menu/grid/tabs/...)</a> - jquery</li>
            <li><a href="/demo/client">Using open source components(bootstrap/jquery-layout/jstree)</a>  - angularjs no typescript
                <ul>
                    <li><a href="/tree/tree.html">jstree</a></li>
                    <li><a href="http://layout.jquery-dev.com/demos/complex.html">jquery-layout</a></li>
                </ul>
            </li>
            <br/>
            <li><a href="/nova-ng">Using off-shelf bootstrap based admin template (bootstrap/jstree/...)</a>  - angularjs no typescript 
                <ul>
                    <li> <a href="http://seantheme.com/source-admin-v1.2/admin/html/index_v2.html">wrapbootstrap</a></li><li> <a href="https://wrapbootstrap.com/theme/source-admin-angular-js-html-apps-WB0N5F2FP">template</a></li>
                </ul>
            </li>
            <li><b><a href="/jq">Using jqwidgets and jstree</a> showing docking/pinning of pannels </b> - angularjs no typescript 
                <ul>
                    <li> <a href="http://www.jqwidgets.com/">jqwidgets</a></li><li> <a href="https://www.jstree.com/">jstree</a></li>
                </ul>
            </li>            
            <br/>
            <li>CSS only sliding panel
                <ul>
                    <li><a href="/tree/pure-drawer/test-squeeze.html">Novina's sliding menu using </a> - <a href="http://mac81.github.io/pure-drawer/">Pure Drawer</a> </li>
                    <li><a href="/tree/">does not integrate well with jstree</a> - plus it is off canvas which is not going to work for our requirement</li>
                    <li><a href="/layout3.html">pegah's own implementation using css only</a></li>
                </ul>
            </li>
        </ul>        
    </h4>

    <h3>Scenario Manager </h3>
    <h4>
        <ul>
            <li><a href="/mxgraph/javascript/examples/scenario.html">Secnario Manager actions using mxgraph</a></li>
            <li><a href="/scenario-manager/jsplumb-test.html">Novina's jsPlumb demo - css animation on nodes</a></li>
            <li><a href="/render.html">MxGraph vs Jsplumb for 1000 connected nodes</a> 
                <ul>
                    <li><a href="/mxgraph/javascript/examples/performance2.html">mxgraph</a></li>
                    <li><a href="/jsplumb-eval/performance.html">jsplumb community</a></li>
                </ul>
            </li>
        </ul>
    </h4>

    <h3>Fonts Demo for RapidReview</h3>
    <h4>
        <ul>
            <li><a href="fonts/demo.html">Using custom fonts</a></li>
            <li><a href="fonts/nofont.html">Using font stacks</a></li>
        </ul>
    </h4>

  </body>
</html>
