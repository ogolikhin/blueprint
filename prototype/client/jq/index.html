<!DOCTYPE html>
<html ng-app="demoApp" lang="en">
<head>
    <title id='Description'>Blueprint</title>
    <link rel="stylesheet" href="jqwidgets-ver3.9.1/jqwidgets/styles/jqx.base.css" type="text/css" />
    <link rel="stylesheet" href="blueprint.min.css" type="text/css" />
    <link rel="stylesheet" href="/demo/client/bower_components/bootstrap/dist/css/bootstrap.css">
    <link rel="stylesheet" href="jstree/themes/default/style.css" />    

    <script src="https://code.jquery.com/jquery-2.1.4.min.js"></script> 
    <!--<script src="https://code.jquery.com/jquery-1.11.3.min.js"></script>-->

    <script type="text/javascript" src="jqwidgets-ver3.9.1/scripts/angular.min.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/scripts/demos.js"></script>
    <link rel="icon" type="image/ico" href="/blueprint.ico"/>



    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxcore.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxbuttons.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxscrollbar.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxdata.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxmenu.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxtooltip.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxlistbox.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxdropdownlist.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxnumberinput.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxradiobutton.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxinput.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxcore.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxribbon.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxlayout.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxdockinglayout.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxtree.js"></script>
    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxwindow.js"></script>

    <script type="text/javascript" src="jqwidgets-ver3.9.1/jqwidgets/jqxangular.js"></script>

        <script src="jstree/jstree.js"></script>


    <style type="text/css">
        .jstree-default .jstree-wholerow-clicked {
            background: #CFDDE9;
            background: -webkit-linear-gradient(top, #CFDDE9 0%, #D1DFEA 100%);    
            background: linear-gradient(to bottom, #CFDDE9 0%, #D1DFEA 100%);
        }


        .jqx-layout-group-auto-hide-content-vertical
        {
            width: 200px;
        }
    </style>
    <script type="text/javascript">
        var demoApp = angular.module("demoApp", ["jqwidgets"]);
        demoApp.controller("demoController", ['$scope', '$window', function ($scope, $window) {
            // the 'layout' JSON array defines the internal structure of the layout
            var layout = [{
                type: 'layoutGroup',
                orientation: 'horizontal',
                items: [

                {
                    type: 'tabbedGroup',
                    width: 220,
                    minWidth: 200,
                    items: [{
                        type: 'layoutPanel',
                        title: 'Explorer',
                        contentContainer: 'SolutionExplorerPanel',
                        initContent: function () {
                            // initialize a jqxTree inside the Solution Explorer Panel
                            var source = [{
                                icon: '/images/earth.png',
                                label: 'Project',
                                expanded: true,
                                items: [{
                                    icon: '/images/Folder.png',
                                    label: 'css',
                                    expanded: true,
                                    items: [{
                                        icon: 'jqwidgets-ver3.9.1/images/nav1.png',
                                        label: 'jqx.base.css'
                                    }, {
                                        icon: 'jqwidgets-ver3.9.1/images/nav1.png',
                                        label: 'jqx.energyblue.css'
                                    }, {
                                        icon: 'jqwidgets-ver3.9.1/images/nav1.png',
                                        label: 'jqx.orange.css'
                                    }]
                                }, {
                                    icon: 'jqwidgets-ver3.9.1/images/folder.png',
                                    label: 'scripts',
                                    items: [{
                                        icon: 'jqwidgets-ver3.9.1/images/nav1.png',
                                        label: 'jqxcore.js'
                                    }, {
                                        icon: 'jqwidgets-ver3.9.1/images/nav1.png',
                                        label: 'jqxdata.js'
                                    }, {
                                        icon: 'jqwidgets-ver3.9.1/images/nav1.png',
                                        label: 'jqxgrid.js'
                                    }]
                                }, {
                                    icon: 'jqwidgets-ver3.9.1/images/nav1.png',
                                    label: 'index.htm'
                                }]
                            }];

                            $('#solutionExplorerTree').jqxTree({ source: source, width: 190 });

                            $('#jstree').jstree({
                              "core" : {
                                "animation" : 0,
                                "check_callback" : true,
                                "themes" : { "stripes" : false },
                                'data' : {
                                  'url' : function (node) {
                                    return node.id === '#' ?
                                      'ajax_demo_roots.json' : 'ajax_demo_children.json';
                                  },
                                  'data' : function (node) {
                                    return { 'id' : node.id };
                                  }
                                }
                              },
                              "types" : {
                                "#" : {
                                  "max_children" : 1, 
                                  "max_depth" : 4, 
                                  "valid_children" : ["root"]
                                },
                                "root" : {
                                  "icon" : "/static/3.2.1/assets/images/tree_icon.png",
                                  "valid_children" : ["default"]
                                },
                                "default" : {
                                  "valid_children" : ["default","file"]
                                },
                                "file" : {
                                  "icon" : "glyphicon glyphicon-file",
                                  "valid_children" : []
                                }
                              },
                              "plugins" : [
                                "contextmenu", "dnd", "search", "state", "types", "wholerow"
                              ]
                            });                            
                        }
                    }, {
                        type: 'layoutPanel',
                        title: 'Properties',
                        contentContainer: 'PropertiesPanel'
                    }]
                },

                {
                    type: 'layoutGroup',
                    orientation: 'vertical',
                    width:  $window.innerWidth - 220 -100,
                    items: [{
                        type: 'documentGroup',
                        height: $window.innerHeight - 200 -60,
                        minHeight: 200,
                        items: [{
                            type: 'documentPanel',
                            title: 'Document 1',
                            contentContainer: 'Document1Panel'
                        }, {
                            type: 'documentPanel',
                            title: 'Document 2',
                            contentContainer: 'Document2Panel'
                        }]
                    }, {
                        type: 'tabbedGroup',
                        height: 200,
                        pinnedHeight: 30,
                        items: [{
                            type: 'layoutPanel',
                            title: 'Error List',
                            contentContainer: 'ErrorListPanel'
                        }, {
                            type: 'layoutPanel',
                            title: 'Output',
                            contentContainer: 'OutputPanel',
                            selected: true
                        }]
                    }]
                },

                {
                    type: 'autoHideGroup',
                    alignment: 'right',
                    width: 100,
                    unpinnedWidth: 200,

                    items: [{
                        type: 'layoutPanel',
                        title: 'Properties',
                        contentContainer: 'Utility-PropertiesPanel'
                    }, {
                        type: 'layoutPanel',
                        title: 'Discussions',
                        contentContainer: 'DiscussionsPanel'
                    }, {
                        type: 'layoutPanel',
                        title: 'Files',
                        contentContainer: 'FilesPanel'
                    }, {
                        type: 'layoutPanel',
                        title: 'Relationships',
                        contentContainer: 'RelationshipsPanel'
                    }, {
                        type: 'layoutPanel',
                        title: 'Outline',
                        contentContainer: 'OutlinePanel'
                    },  {
                        type: 'layoutPanel',
                        title: 'Browse',
                        contentContainer: 'BrowsePanel'
                    }, {
                        type: 'layoutPanel',
                        title: 'History',
                        contentContainer: 'HistoryPanel'
                    }, {
                        type: 'layoutPanel',
                        title: 'Review',
                        contentContainer: 'ReviewsPanel'
                    }]


                }

                ]
            }];




            $scope.settings = {width: $window.innerWidth , height: $window.innerHeight -60 , layout: layout, theme: 'blueprint' };

        }]);


    </script>




</head>
 <body ng-controller="demoController" style="margin:0;overflow:hidden;">
    <jqx-menu jqx-width="'100%'" jqx-height="'60px'" jqx-animation-show-duration="0" jqx-animation-hide-duration="0" jqx-animation-show-delay="0" jqx-animation-hide-delay="0" jqx-theme="'blueprint'">
      <div id="blue" style="display: block; color: white; position: absolute; top: 0px; height: 28px; width: 100%; background-color: rgb(28, 41, 92);">
        <span style="position: fixed; left: 1em; top: 2px; font-family: 'Hind', sans-serif; font-size: large"><b>blueprint ®</b></span>
        <span style="position: fixed; right: 1em; top: 6px;"><span class="glyphicon glyphicon-user"></span> Welcome Pegah Tabrizi | <span class="glyphicon glyphicon-home"></span> | Logout</span>
      </div>  
      <ul style="margin-top:29px;">
        <li><a href="#Home"><span class="glyphicon glyphicon-home"></span></a>
            <ul>
              <li>Save
                <ul>
                  <li>Save All</li>
                  <li>Save</li>
                </ul>
              </li>
              <li>Publish
                <ul>
                  <li>Publish All</li>
                  <li>Publish</li>
                </ul>
              </li>
              <li>Discard
                <ul>
                  <li>Discard All</li>
                  <li>Discard</li>
                </ul>
              </li>
              <li>Refresh
                <ul>
                  <li>Refresh All</li>
                  <li>Refresh</li>
                </ul>
              </li>
              <li>Print
                <ul>
                  <li>Print to PDF</li>
                  <li>Print to Word</li>
                </ul>
              </li>
              <li>Open</li>
              <li>Open Recent
                <ul>
                  <li>Project 1</li>
                  <li>Project 2</li>
                </ul>
              </li>
              <li>Close
                <ul>
                  <li>Project 1</li>
                  <li>Project 2</li>
                </ul>
              </li>                             
              <li>Manage
                <ul>
                  <li>Jobs</li>
                  <li>Instance Administration</li>
                  <li>Project 1 - Project Administration</li>
                </ul>
              </li>                           
            </ul>
          </li>
          <li>
           Home
            <ul>
              <li>TODO</li>
            </ul>
          </li>
          <li>
            View
            <ul>
              <li>TODO</li>
            </ul>
          </li>
          <li>
            Import/Export
            <ul>
              <li>Import
                <ul>
                  <li>Artifacts</li>
                  <li class="disabled">Glossary Terms</li>
                  <li>Visio Diagram</li>
                  <li>Excel Update</li>
                </ul>
              </li>
              <li>Export
                <ul>
                  <li>Office Document
                    <ul>
                      <li>From Selection</li>
                      <li>From Selection - include descendants</li>
                    </ul>
                  </li>
                  <li>ALM Export</li>
                  <li>Test Plans</li>
                  <li>Artifact List View</li>
                  <li>Generate Excel Update Template</li>
                  <li class="disabled">Visio Diagram</li>
                </ul>
              </li>
            </ul>
          </li>

          <li>Help
            <ul>
              <li>Blueprint &trade; Help</li>
              <li>Keyboard Shortcuts</li>
               <li type='separator'></li>
              <li>About Blueprint ®</li>
            </ul>
          </li>         
      </ul>
    </jqx-menu>


       <jqx-docking-layout jqx-settings="settings" id="jqxLayout">
        <!--The panel content divs can have a flat structure-->


        <!--right tabbedGroup-->
        <div data-container="SolutionExplorerPanel">
            <!--<div jstree id="jstree" selected-node="selectedTreeNode" selected-path ="selectedPath" selected-node-changed="nodeChanged" style="z-index:20000;"> </div> -->
            <div id="jstree"></div>

        </div>
        <div data-container="PropertiesPanel">
            
          <div id="solutionExplorerTree" style="border: none;"></div> 

        </div>

        <!--documentGroup-->
        <div data-container="Document1Panel">
          <jqx-grid jqx-settings="gridsettings"></jqx-grid>
          <br />
          {{bindingCompleted}}

        </div>
        <div data-container="Document2Panel">
            Document 2 content</div>
        <!--bottom tabbedGroup-->
        <div data-container="ErrorListPanel">
            List of errors</div>
        <div data-container="OutputPanel">
            Output</div>

        <!--autoHideGroup-->
        <div data-container="Utility-PropertiesPanel">Properties</div>
        <div data-container="DiscussionsPanel">List of Discussions</div>
        <div data-container="FilesPanel">Files</div>
        <div data-container="RelationshipsPanel">Relationships</div>
        <div data-container="OutlinePanel">Outline</div>
        <div data-container="BrowsePanel">Browse</div>
        <div data-container="HistoryPanel">History</div>
        <div data-container="ReviewsPanel">Review</div>

    </jqx-docking-layout>
</body>
</html>
