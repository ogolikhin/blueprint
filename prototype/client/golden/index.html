<!DOCTYPE html>
<!--[if lt IE 7]>      <html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js"> <!--<![endif]-->
<head>
  <title> Blueprint HTML5 SPA Prototype - Using Golden Layout</title>
  <link rel="icon" type="image/ico" href="/blueprint.ico"/>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">

  <!-- bower components -->
  <script type="text/javascript" src="client/bower_components/jquery/dist/jquery.min.js"></script>
  <script type="text/javascript" src="client/bower_components/golden-layout/dist/goldenlayout.min.js"></script>
  <script type="text/javascript" src="client/bower_components/bootstrap/dist/js/bootstrap.min.js"></script>

  <script type="text/javascript" src="client/bower_components/angular/angular.min.js"></script>

  <script type="text/javascript" src="client/bower_components/angular-bootstrap/ui-bootstrap.min.js"></script>
  <script type="text/javascript" src="client/bower_components/angular-bootstrap/ui-bootstrap-tpls.min.js"></script>


  <link type="text/css" rel="stylesheet" href="client/bower_components/golden-layout/src/css/goldenlayout-base.css" />
  <link type="text/css" rel="stylesheet" href="client/bower_components/golden-layout/src/css/goldenlayout-dark-theme.css" />
  <link type="text/css" rel="stylesheet" src="client/bower_components/bootstrap/dist/css/bootstrap-theme.min.css" />
  <!-- bower -->




  <link type="text/css" rel="stylesheet" href="styles/golden.css" />

</head>

<body>
  <link href='http://fonts.googleapis.com/css?family=Hind:400,500' rel='stylesheet' type='text/css'>

<div id="blue" style="display: block; color: white; position: absolute; top: 0px; height: 28px; width: 100%; background-color: rgb(28, 41, 92);">
              <span style="font-family: 'Hind', sans-serif; font-size: large">&nbsp;&nbsp;&nbsp;<b>blueprint Â®</b></span>
              <span style="position: fixed;right: 15px; top:6px;">Welcome Pegah Tabrizi | Home | Logout</span>

 </div>         
 <br/><br/>

<div id="wrapper">
  <div id="menuContainer">
    
<!-- Single button -->


  </div>


  <div id="layoutContainer"></div>
</div>


<template type="text/html" id="userlistTemplate">
  <ul ng-controller="userlistController" class="userlist">
    <li ng-repeat="user in users" 
        ng-click="select( user )"
        ng-class="{selected:user.isSelected}">
      {{user.name}}
    </li>
  </ul>
</template>

  <template type="text/html" id="userDetailTemplate">
    <div ng-controller="userdetailsController" class="userdetails">
      <img src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/152047/{{user.img}}" width="100" height="100" />
      <h2>{{user.name}}</h2>
      <p>{{user.street}}</p>
    </div>
</template>

<script>
angular.module('userlist', [] )
  .controller('userlistController', function( $scope, $timeout, container, state ) {

    var selectedUser = {};

    $scope.users = [
      { name: 'Jackson Turner', street: '217 Tawny End', img: 'men_1.jpg' },
      { name: 'Megan Perry', street: '77 Burning Ramp', img: 'women_1.jpg' },
      { name: 'Ryan Harris', street: '12 Hazy Apple Route', img: 'men_2.jpg' },
      { name: 'Jennifer Edwards', street: '33 Maple Drive', img: 'women_2.jpg' },
      { name: 'Noah Jenkins', street: '423 Indian Pond Cape', img: 'men_3.jpg' }
    ];

    $timeout(function(){
      $scope.select( $scope.users[ state.selectedUserIndex ] );
    });

    $scope.select = function( user ) {
      selectedUser.isSelected = false;
      user.isSelected = true;
      selectedUser = user;
      container.extendState({ selectedUserIndex: $scope.users.indexOf( user ) });
      container.layoutManager.eventHub.emit( 'userSelected', user );
    };
  });

angular.module('userdetails', [] )
  .controller('userdetailsController', function( $scope, container, state ) {
    $scope.user = state.user || null;

    container.layoutManager.eventHub.on( 'userSelected', function( user ){
      $scope.user = user;
      container.extendState({ user: user });
      $scope.$apply();
    });
  });


var AngularModuleComponent = function( container, state ) {
  var html = $( '#' + state.templateId ).html(),
    element = container.getElement();
  
  element.html( html );

  angular
    .module( state.module )
    .value( 'container', container )
    .value( 'state', state );

  angular.bootstrap( element[ 0 ], [ state.module ] );
};

var myLayout = new GoldenLayout({
  content:[{
    type: 'row',
    content: [{
      width: 20,
      title: 'Explorer',
      type: 'component',
      componentName: 'angularModule',
      componentState: {
        module: 'userlist',
        templateId: 'userlistTemplate',
        selectedUserIndex: 2
      }
      },{
      type: 'component',
      title: 'Selected User',
      componentName: 'angularModule',
      componentState: {
        module: 'userdetails',
        templateId: 'userDetailTemplate'
      }
    },{
      type: 'component',
      title: 'Selected User',
      componentName: 'angularModule',
      componentState: {
        module: 'userdetails',
        templateId: 'userDetailTemplate'
      }
    }]
  }]
} , $('#layoutContainer') );

myLayout.registerComponent( 'angularModule', AngularModuleComponent );
myLayout.init();
</script>
</body>
</html>